libs/wingc/src/type_check.rs

```diff
@@ -2408,14 +2410,14 @@ impl<'a> TypeChecker<'a> {
 	}
 
 	fn type_check_arg_list_against_function_sig(
-		&mut self,
-		arg_list: &ArgList,
-		func_sig: &FunctionSignature,
-		call_span: &impl Spanned,
-		arg_list_types: ArgListTypes,
+		&mut self, // TypeChecker
+		arg_list: &ArgList, // Arg List exprs
+		func_sig: &FunctionSignature, // Function Signature
+		call_span: &impl Spanned, // Function call position
+		arg_list_types: ArgListTypes, // Arg List types belonging to arg list exprs
```

Comment:
:+1: 
One thing that seems a bit surprising here is that we have this `arg_list_types` which I can't really account for since the information should already be stored in `arg_list` itself. `arg_list` is a collection of exprs, as you mention, and all these exprs are already type checked so you can, theoretically just use `get_expr_type` on them. @MarkMcCulloh any idea why the seemingly redundant arg here?

Reply:
Also not that this whole thing is used when type-checking a function call, and not a function type's match to an expected signature. So I think this isn't related to [#2230](https://github.com/winglang/wing/issues/2230).

Reply:
I noticed the variadic args are type validated in here, thus in theory a variadic function arg passed to the function call could be related to #2230 I thought